homeassistant/components/mqtt/__init__.py

```diff
@@ -364,18 +364,27 @@ async def finish_dump(_):
     hass.data[DATA_CONFIG_ENTRY_LOCK] = asyncio.Lock()
     hass.data[CONFIG_ENTRY_IS_SETUP] = set()
 
-    async def async_forward_entry_setup():
+    async def async_forward_entry_setup(config_entry):
```

Comment:
Rename this function and let it setup discovery too after gathering setup of all platforms, otherwise discovery may be setup before the platforms